### PR TITLE
fix: change rate limit value as a number

### PR DESCRIPTION
**Pull request by nguyentoanit**

### CODE REVIEW

**Comment by moltar** [comment]
Task linked: [GHORG-65 Get rate limit value from header](https://app.clickup.com/t/2250003/GHORG-65)

**Comment by nguyentoanit** [comment]
Resolve comment: https://github.com/ScaleLeap/selling-partner-api-sdk/pull/112#discussion_r634897061

File: src/types/errors/selling-partner-api-errors.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable max-classes-per-file */
+import toNumber from 'lodash.tonumber'
```
Comment (moltar) [inline src/types/errors/selling-partner-api-errors.ts:2]: Why do we need this? Why not just do `Number("123")` or `parseFloat("123.456")`?

**Review by moltar — CHANGES_REQUESTED** [review]

File: src/types/errors/selling-partner-api-errors.ts [hunk 2]
```diff
@@ -47,11 +47,14 @@ export class SellingPartnerNotFoundError extends SellingPartnerGenericError {}
...
 export class SellingPartnerUnsupportedMediaTypeError extends SellingPartnerGenericError {}
 export class SellingPartnerTooManyRequestsError extends SellingPartnerGenericError {
-  public rateLimit?: string
+  public rateLimit?: number
 
   public constructor(error: ModelError, headers: Headers) {
     super(error, headers)
-    this.rateLimit = headers['x-amzn-RateLimit-Limit'] || headers['x-amzn-ratelimit-limit']
+    this.rateLimit =
+      Number(headers['x-amzn-RateLimit-Limit']) ||
```
Comment (moltar) [inline src/types/errors/selling-partner-api-errors.ts:55]: We need to guard here against `NaN`. I recommend to create a small helper function to deal with this.

`Number(undefined)` = `NaN`
`Number("foo")` = `NaN`

We certainly don't want any NaNs :)

**Review by moltar — CHANGES_REQUESTED** [review]

File: src/types/errors/selling-partner-api-errors.ts (hunk 2 above)
Comment (nguyentoanit) [reply src/types/errors/selling-partner-api-errors.ts:55]: > We certainly don't want any NaNs :)

```
NaN || NaN || undefined // undefined
```

@moltar : 
logical OR (||) operator against `NaN` already? 🤔 
I don't think we need a helper function.

Comment (moltar) [reply src/types/errors/selling-partner-api-errors.ts:55]: Good point 😁 👍🏼

**Review by moltar — APPROVED** [review]

**Pull request merged**